dronegen/main.go

```diff
@@ -0,0 +1,122 @@
+package main
+
+import (
+	"bytes"
+	"fmt"
+	"io/ioutil"
+	"os"
+
+	"gopkg.in/yaml.v2"
+)
+
+func main() {
+	if err := checkDroneCLI(); err != nil {
+		fmt.Println(err)
+		os.Exit(1)
+	}
+	var pipelines []pipeline
+
+	pipelines = append(pipelines, testPipelines()...)
+	pipelines = append(pipelines, pushPipelines()...)
+	pipelines = append(pipelines, tagPipelines()...)
+	pipelines = append(pipelines, cronPipelines()...)
+	pipelines = append(pipelines, promoteBuildPipeline())
+	pipelines = append(pipelines, updateDocsPipeline())
+
+	if err := writePipelines(".drone.yml", pipelines); err != nil {
+		fmt.Println("failed writing drone pipelines:", err)
+		os.Exit(1)
+	}
+
+	if err := signDroneConfig(); err != nil {
+		fmt.Println("failed signing .drone.yml:", err)
+		os.Exit(1)
+	}
+}
+
+func writePipelines(path string, newPipelines []pipeline) error {
```

Comment:
I've also run into an issue where I accidentally duplicated a step in a pipeline and the drone's error message is not super helpful:
```
Linter: duplicate step name
```
it even had the wrong pipeline name in the front. I was wondering whether you guys considered adding basic linting functionality prior to writing pipelines to disk? `drone lint` alone is not helpful - it fails locally for multiple reasons but I've seen linting implemented as part of [boilr-runner](https://github.com/drone/boilr-runner) (which we're using as well somewhere, probably indirectly).

Reply:
It sounds like a good idea to check that the list of output pipeline names doesn't have any duplicates.